Update redundancy.md (#4558)
MicrosoftDocs/architecture-center: docs/guide/design-principles/redundancy.md

## Build redundancy into your application, to avoid having single points of failure

A resilient application routes around failure. Identify the critical paths in your application. Is there redundancy at each point in the path? When a subsystem fails, will the application fail over to something else?
 
In a perfect implementation, adding uniform redundancy could exponentially increase your system's availability. For example, imagine you have `N` equivalent, equally balanced components which:

- can malfunction independently and simultaneously removed from the pool.
- have identical state or no state.
- have no work in progress that is permanently lost during the malfunction.
- are identical in capabilities.
- have no dependencies on each other.

> - handles the reduction of capacity without additional malfunction

Reviewer: ```suggestion
- handles the reduction of capacity without additional malfunction.
```

> - can malfunction independently and simultaneously removed from the pool.
> - have identical state or no state.
> - have no work in progress that is permanently lost during the malfunction.
> - are identical in capabilities.
> - have no dependencies on each other.
> - handles the reduction of capacity without additional malfunction.

Reviewer 2: ```suggestion
- can malfunction independently and simultaneously removed from the pool
- have identical state or no state
- have no work in progress that is permanently lost during the malfunction
- are identical in capabilities
- have no dependencies on each other
- handles the reduction of capacity without additional malfunction
```

> If each individual component has an availability of `A`, then the overall system availability can be calculated using the formula `1 - (1 - A)<sup>N</sup>`.

Reviewer: ```suggestion
If each individual component has an availability of `A`, then the overall system availability can be calculated using the formula `1 - (1 - A)^N`.
```

## Build redundancy into your application, to avoid having single points of failure

A resilient application routes around failure. Identify the critical paths in your application. Is there redundancy at each point in the path? When a subsystem fails, will the application fail over to something else?
 

> If designed properly, adding redundancy can exponentially increase your system’s availability. If you have ```N``` hosts which fail independently (e.g. by not being within the same datacenter), and each host has availability of ```A```, the overall system availability can be calculated using the formula ```1 - (1 - A) ^ N```. 

Reviewer: ```suggestion
In a perfect implementation, adding uniform redundancy could exponentially increase your system's availability. For example, imagine you have `N` equivalent, equally balanced components which:

- can malfunction independently and simultaneously removed from the pool.
- have identical state or no state.
- have no work in progress that is permanently lost during the malfunction.
- are identical in capabilities.
- have no dependencies on each other.
- handles the reduction of capacity without additional malfunction

If each individual component has an availability of `A`, then the overall system availability can be calculated using the formula `1 - (1 - A)<sup>N</sup>`.
```

**Partition for availability**. Database partitioning is often used to improve scalability, but it can also improve availability. If one shard goes down, the other shards can still be reached. A failure in one shard will only disrupt a subset of the total transactions.

> **Test and validate your redundant components**. Reliability benefits in many ways from simplicity and adding redundancy can increase complexity. To ensure that adding redundancy actually leads to higher availability, you should validate followings: 

Reviewer: ```suggestion
**Test and validate your redundant components**. Reliability benefits in many ways from simplicity and adding redundancy can increase complexity. To ensure that adding redundancy actually leads to higher availability, you should validate following:

```

**Include redundancy for your routing solution**. Consider designing a [Global routing redundancy solution](../networking/global-web-applications/overview.md) for mission-critical web applications.

> **Test and validate your redundant components**. Adding redundancy increases complexity which in turn reduces availability. To ensure that adding redundancy actually leads to higher availability, you should test and validate followings: 

Reviewer: ```suggestion
**Test and validate your redundant components**. Reliability benefits in many ways from simplicity and adding redundancy can increase complexity. To ensure that adding redundancy actually leads to higher availability, you should validate followings: 
```

> **Test and validate your redundant components**. Adding redundancy increases complexity which in turn reduces availability. To ensure that adding redundancy actually leads to higher availability, you should test and validate followings: 

Reviewer: Is this really a "multi-region" bullet?  I don't see anything about region in here. Should this instead be the last bullet point in the Recommendations H2 instead of the last bullet point in the Multi-region H3?

Author: Relocated it

> * Given the added complexity, does your overall system achieve a net-positive improvement in availability?
> * Can your system reliably detect healthy and unhealthy redundant components?
> * Can your system reliably scale out and scale-in the redundant components? This sounds easy but requires testing and validation.

Reviewer: ```suggestion
- Can your system *reliably* detect healthy and unhealthy redundant components, and safely and expeditiously remove them from the component pool?
- Can your system *reliably* scale out and in the redundant components?
- Can your routine, ad hoc, and emergency workload operations handle the redundancy?
```

> 

Reviewer: ```suggestion
```
